src/test/java/io/temporal/workflow/MetricsTest.java

```diff
@@ -335,4 +354,63 @@ public WorkflowInvoker interceptExecuteWorkflow(
       return new BaseWorkflowInvoker(i, next);
     }
   }
+
+  @Test
+  public void testTemporalFailureMetric() throws InterruptedException {
+    setUp(
+        com.uber.m3.util.Duration.ofMillis(300),
+        WorkerFactoryOptions.newBuilder()
+            .setWorkflowInterceptor(new CorruptedSignalWorkflowInterceptor())
+            .build());
+
+    try {
+      WorkflowServiceStubs serviceStubs =
+          testEnvironment.getWorkflowClient().getWorkflowServiceStubs();
+
+      serviceStubs.blockingStub().describeNamespace(DescribeNamespaceRequest.newBuilder().build());
+      fail("failure expected");
+    } catch (StatusRuntimeException e) {
+      assertEquals(Status.Code.UNIMPLEMENTED, e.getStatus().getCode());
+    }
+
+    // Wait for reporter
+    Thread.sleep(600);
```

Comment:
Where does 600 come from? Is this long enough for most/all developers who will run these?

Reply:
Changed to more reasonable value.

Reply:
It looks like a smaller value leads to intermittent test failures. So keeping it at 600.